api/internal/localizer/localizer_test.go

```diff
@@ -95,6 +113,29 @@ func reportFSysDiff(t *testing.T, fSysExpected filesys.FileSystem, fSysActual fi
 	require.NoError(t, err)
 }
 
+func checkDeprecationWarning(t *testing.T) {
```

Comment:
This checks that we warn against the use of deprecated fields, which we currently do because we call `kusttarget.Load`. Do we want to keep this behavior in `localize`, including when we write our own legacy `Load`? 

If we don't want `localize` to be responsible for warning users and we don't care that our test output is polluted by warnings, I can remove this check.

Reply:
IMO `localize` should not warn users about deprecated fields, because `build` will do that for them and we should minimize unnecessary warnings.

You can include your own `Load` in this PR (or another very small PR if you wish), it doesn't need to much besides calling `loadKustFile`, `Unmarshal`, and `FixKustomizationPostUnmarshalling`, so it should be a very small change.

Reply:
Sounds good! Done. I will follow-up with a small `Load` PR, as suggested.

Reply:
`Load` implemented in #4918